lab

```diff
@@ -0,0 +1,211 @@
+                echo "PASSED!"
+                if [ "$scenario" == "1" ]; then
+                    echo -e "\nCOMPLETION CODE: 0msk8e"
+                fi
+                if [ "$scenario" == "2" ]; then
+                    echo -e "\nCOMPLETION CODE: geg3my"
+                fi
+                if [ "$scenario" == "3" ]; then
+                    echo -e "\nCOMPLETION CODE: 7te0hl"
+                fi
+                if [ "$scenario" == "4" ]; then
+                    echo -e "\nCOMPLETION CODE: mj8ra8"
+                fi
+                if [ "$scenario" == "5" ]; then
+                    echo -e "\nCOMPLETION CODE: ax6dnw"
+                fi
+                if [ "$scenario" == "6" ]; then
+                    echo -e "\nCOMPLETION CODE: szj8ta"
+                fi
+                if [ "$scenario" == "7" ]; then
+                    echo -e "\nCOMPLETION CODE: ajkm9o"
+                fi
+                if [ "$scenario" == "8" ]; then
+                    echo -e "\nCOMPLETION CODE: vzo8ll"
+                fi
+                if [ "$scenario" == "9" ]; then
+                    echo -e "\nCOMPLETION CODE: ktn7ao"
+                fi
+            else
+                echo "FAILED!"
+            fi
+            ;;
+        fix)
+            if [ $rc -eq 0 ]; then
+                echo "Scenario $scenario has been fixed!"
+            else
+                echo "Failed to fix scenario $scenario"
+            fi
+            ;;
+    finish)
+            if [ $rc -eq 0 ]; then
+                echo "Scenario $scenario has been cleaned up."
+            else
+                echo "Failed to clean up after scenario $scenario"
+            fi
+            ;;
+    esac
+}
+
+
+function call_ansible() {
+    pushd $ROLES_DIR >> /dev/null
+    $ANSIBLE_CMD -t $1 -e scenario=$2 $ROLES_DIR/playbooks/scenario.yml >> /tmp/ansible_scenario_output.txt
+    rc=$?
+    popd >> /dev/null
+
+    return $rc
+}
+
+
+function check_scenario_parameter() {
+    local scenario=$1
+
+    if [ "x$scenario" == "x" ]; then
+        echo "Missing scenario number"
+        echo $HELP
+        exit $E_BAD_INPUT
+    fi
+
+    if [ ! -e $ROLES_DIR/roles/scenario$scenario ]; then
+        errcho "Trying to work with scenario number that is not implemented: scenario $scenario does not exist"
+        exit $E_SCENARIO
+    fi
+
+    if [ -e $WORKIND_DIR/.scenario[0-9] -a ! -e $WORKIND_DIR/.scenario$scenario ]; then
+        existing_scenario=$(ls -d $WORKIND_DIR/.scenario[0-9] | sed 's/.*\.scenario\([0-9]*\)/\1/g')
+        errcho "There is deployed an existing scenario $existing_scenario. Please run the cleanup by issuing following command: $0 finish $existing_scenario"
+        exit $E_SCENARIO
+    fi
+}
+
+
+function checks() {
+    local action=$1
+    local scenario=$2
+
+    case "$action" in
+        fix|grade|finish)
+            check_scenario_parameter $scenario
+            if [ "x$scenario" != "x" -a ! -e $WORKIND_DIR/.scenario$scenario ]; then
```

Comment:
`"x$scenario" != "x"` can't ever be false because the check_scenario_parameter would have exited. I think it can be removed, no?